rust/xaynet-client/src/mobile_client/mod.rs

```diff
@@ -85,15 +85,22 @@ impl MobileClient {
         Ok(global_model)
     }
 
-    pub fn perform_task(self) -> Result<Self, MobileClientError> {
+    pub fn perform_task(self) -> Result<Self, (Self, MobileClientError)> {
+        let mut runtime = match Self::runtime() {
+            Ok(runtime) => runtime,
+            // We don't want to loose the current client because of a runtime error.
+            // Therefore we return the error as well as the current client.
+            Err(err) => return Err((self, err.into())),
```

Comment:
Mmh I understand why we're doing that but it feels awkward. I think it would at least deserve an explanation in the docstring.

Reply:
I added the documentaion for all public methods :
In addtion, I renamed two methods `deserialize` -> `restore` and `perfrom_task` -> `try_to_proceed`.